src/basicfuns.jl

```diff
@@ -44,6 +44,31 @@ function xlog1py(x::Number, y::Number)
     return iszero(x) && !isnan(y) ? zero(result) : result
 end
 
+"""
+$(SIGNATURES)
+
+Return `x * exp(x)` for `x > -Inf`, or zero if `x == -Inf`.
+
+```jldoctest
+julia> xexpx(-Inf)
+0.0
+```
+"""
+xexpx(x::Real) = isfinite(x) ? x * exp(x) : exp(x)
```

Comment:
The check is a bit simpler and more direct (since currently the second bramch is used not only for `-Inf`) if you write

```suggestion
function xexpx(x::Real)
    expx = exp(x)
    iszero(expx) ? expx : x * expx
end
```

Reply:
I think the original code matches the intention very well though, so I would be OK with it as is.

Reply:
Honestly I also prefer the original version :)

Reply:
I still think `iszero` is clearer since it focuses on the one special case we want to handle (similar to the `iszero` in eg `xlogx`). `isfinite` catches other cases as well and is a more complicated check, for `AbstractFloat` (https://github.com/JuliaLang/julia/blob/fe1264a5c980f9a17be715f402fce592c7f6f35c/base/float.jl#L499) and in particular generic `Real` (https://github.com/JuliaLang/julia/blob/fe1264a5c980f9a17be715f402fce592c7f6f35c/base/float.jl#L500).